datastore/longhorn.go

```diff
@@ -585,29 +585,73 @@ func checkEngine(engine *longhorn.Engine) error {
 	return nil
 }
 
-// PickVolumeCurrentEngine pick the current Engine from the Engine list of a volume with the given namespace
-func (s *DataStore) PickVolumeCurrentEngine(v *longhorn.Volume, es map[string]*longhorn.Engine) (*longhorn.Engine, error) {
-	if len(es) == 0 {
-		return nil, nil
+// GetCurrentEngineAndExtras pick the current Engine and extra Engines from the Engine list of a volume with the given namespace
+func GetCurrentEngineAndExtras(v *longhorn.Volume, es map[string]*longhorn.Engine) (currentEngine *longhorn.Engine, extras []*longhorn.Engine, err error) {
+	if len(es) == 1 {
+		for _, e := range es {
+			return e, nil, nil
+		}
 	}
```

Comment:
Should we remove this part? but just rely on the below new loop?

Reply:
I am not sure if there are any unexpected cases that would lead to the only engine does not set `Spec.Active`. Let me add a warn log for this case first... It should be like:
```
if len(es) == 1 {
	for _, e := range es {
		if !e.Spec.Active {
			log.Warn("xxxxx")
		}
		return e, nil, nil
	}
}
```

Reply:
I mean probably u can just remove this if block, and just leave the following new logic to handle.

Reply:
I mean, if I remove this block, this function may error out once somehow the only engine of a volume does not set `Spec.Active`. Then the reconcile loop will error out. This is what I want to avoid.

Reply:
Got your point, just for `in case`, so the concern is `I am not sure if there are any unexpected cases that would lead to the only engine does not set Spec.Active`. .

Reply:
YES. Erroring out would implicitly force users to report the issue once it's really triggered. It's reasonable and good for us but may annoy users. If it's better to expose the issue as early as possible, I am fine with removing this block. What do you think?

Reply:
Then, removed 🚀